src/main/java/com/lohika/morning/ecs/domain/template/TemplateEditView.java

```diff
@@ -0,0 +1,126 @@
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import com.vaadin.data.BeanValidationBinder;
+import com.vaadin.data.Binder;
+import com.vaadin.data.HasValue;
+import com.vaadin.event.ShortcutAction;
+import com.vaadin.icons.VaadinIcons;
+import com.vaadin.navigator.View;
+import com.vaadin.navigator.ViewChangeListener;
+import com.vaadin.spring.annotation.SpringView;
+import com.vaadin.ui.Button;
+import com.vaadin.ui.CheckBox;
+import com.vaadin.ui.ComboBox;
+import com.vaadin.ui.FormLayout;
+import com.vaadin.ui.HorizontalLayout;
+import com.vaadin.ui.RichTextArea;
+import com.vaadin.ui.TextField;
+import com.vaadin.ui.themes.ValoTheme;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+
+import static java.lang.Math.abs;
+import static java.util.Arrays.asList;
+
+@Slf4j
+@RequiredArgsConstructor
+@SpringView(name = TemplateEditView.VIEW_NAME)
+public class TemplateEditView extends HorizontalLayout implements View {
+  public static final String VIEW_NAME = "addTemplate";
+  private static final List<Integer> PERIOD_ITEMS = asList(-14, -7 - 5, 0, 1, 7, 14);
+
+  private final TextField name = new TextField("Name");
+  private final TextField subject = new TextField("Subject");
+  private final RichTextArea body = new RichTextArea("Body");
+  private final ComboBox<Integer> priority = new ComboBox<>("Select priority", PERIOD_ITEMS);
+  private final CheckBox attendee = new CheckBox("For all attendee");
+  private final TextField emails = new TextField("Emails");
+
+  private final Button saveButton = new Button("Save", VaadinIcons.CHECK);
+  private final Button deleteButton = new Button("Delete", VaadinIcons.TRASH);
+  private final Button cancelButton = new Button("Cancel");
+  private final Binder<Template> validation = new BeanValidationBinder<>(Template.class);
+
+  private final TemplateService templateService;
+
+  @PostConstruct
+  public void init() {
+
+    HorizontalLayout buttons = new HorizontalLayout(saveButton, deleteButton, cancelButton);
+
+    FormLayout form = new FormLayout(name, subject, body, attendee, emails, priority, buttons);
+    addComponents(form);
+
+    validation.bindInstanceFields(this);
+
+    attendee.addValueChangeListener(this::onAttendeeChange);
+
+    priority.setItemCaptionGenerator(this::generatePriorityCaption);
+    priority.setEmptySelectionAllowed(false);
+
+    saveButton.setStyleName(ValoTheme.BUTTON_PRIMARY);
+    saveButton.setClickShortcut(ShortcutAction.KeyCode.ENTER);
+    saveButton.addClickListener(this::createTemplate);
+
+    deleteButton.addClickListener(this::deleteTemplate);
+    deleteButton.setEnabled(isDeleteEnabled());
+
+    cancelButton.setClickShortcut(ShortcutAction.KeyCode.ESCAPE);
+    cancelButton.addClickListener(this::cancelTemplate);
+  }
+
+  private boolean isDeleteEnabled() {
+    return validation.getBean() != null && validation.getBean().getId() != null;
```

Comment:
If you call this method only from the `enter()` (i.e. after `binder.setBean()`), the `binder.getBean()` should not be null, right?

Reply:
rename to `isEntityPersisted()`